packages/react-i18n-webpack-plugin/src/react-i18n-webpack-plugin.ts

```diff
@@ -0,0 +1,264 @@
+            .for('CallExpression')
+            .tap(PLUGIN_NAME, (originalExpression: CallExpression) => {
+              if (
+                parser.state.module.resource.indexOf('node_modules') !== -1 &&
+                !parser.state.module.resource.endsWith('tsx') &&
+                !parser.state.i18nImports
+              ) {
+                return;
+              }
+
+              const componentPath = parser.state.module.resource;
+              const componentDir = parser.state.module.context;
+              const identifierName = parser.state.i18nImports.get(
+                componentPath,
+              );
+
+              if (
+                !identifierName ||
+                originalExpression.callee.type !== 'Identifier'
+              ) {
+                return;
+              }
+
+              let expression: CallExpression | null = null;
+              if (originalExpression.callee.name === 'compose') {
+                const foundExpressions: Array<
+                  Expression | SpreadElement
+                > = originalExpression.arguments.filter(
+                  node =>
+                    node.type === 'CallExpression' &&
+                    node.callee.type === 'Identifier' &&
+                    node.callee.name === identifierName,
+                );
+
+                if (foundExpressions.length > 0) {
+                  expression = foundExpressions[0] as CallExpression;
+                }
+              } else if (originalExpression.callee.name === identifierName) {
+                expression = originalExpression;
+              }
+
+              // skip calls where consumer manually added arguments
+              if (!expression || expression.arguments.length > 0) {
+                return;
+              }
+
+              const translationFiles = getTranslationFiles(parser);
+              if (translationFiles.length === 0) {
+                return;
+              }
+
+              // check if fall back exist
+              const fallBackExist = translationFiles.find(
+                translationFile =>
+                  translationFile === `${this.options.fallbackLocale}.json`,
+              );
+
+              const fallBackFileRelativePath = path.join(
+                './',
+                TRANSLATION_DIRECTORY_NAME,
+                `${this.options.fallbackLocale}.json`,
+              );
+
+              if (!fallBackExist) {
+                compilation.errors.push(
+                  `${componentPath}\n` +
+                    `${identifierName} 's arguments was not automatically filled in because` +
+                    `fallback translation file was not found at ${fallBackFileRelativePath} \n`,
+                );
+                return;
+              }
+
+              // Add a top-level fallbackLocale import
+              const fallbackLocaleID = camelCase(this.options.fallbackLocale);
+
+              const fallbackFileExpression = ParserHelpers.requireFileAsExpression(
```

Comment:
This seems to create a `require` statement, which I *think* doptimizes some things during bundling. Can it use a default import instead?

Reply:
maybe... 😹 just need to figure out how
but also, do we not want the optimization here?

Reply:
We never want to deoptimize the way the module is compiled.

Reply:
Didnt really resolved this yet. But it does seem it will take some time to research. Going to concentrate on shipping this first and do this as a improvement toward the end.